src/parse.rs

```diff
@@ -251,13 +256,28 @@ impl<'a> Parser<'a> {
         let mut end = ix + 1 + codepoint_len(b);
         let mut size = 1;
         if is_digit(b) {
+            // don't allow numbered groups if named groups have been used
+            if self.named_groups.len() > 0 {
+                return Err(Error::NamedBackrefOnly);
+            }
```

Comment:
I think we need to check this at the end of parsing, otherwise this would not be detected as a syntax error:

`(a)\1(?<name>b)`

Reply:
Thank you for all of the help. I added a test for this and now check for mixed numeric / named backrefs at the end of parsing. Unfortunately it means I had to go through the re again rather than detect the issue as it was being parsed the first time, but I couldn't think of another way without beginning to work outside of parser.rs.